### ISSUE TITLE
[Feature Request] Configurable layout for terminal joplin

### ISSUE THREAD

**Issue by sbussetti** [laurent22/joplin#2110]
IWBTAT be able to configure the layout of the panes in terminal mode.  Arranging panes would be amazing (horizontal vs vertical splits), but even just being able to set the COLs and ROWs for each would be awesome.

**Comment by laurent22** [comment]
> IWBTAT

what?

But otherwise yes that would be a useful feature.

**Comment by laurent22** [comment]
**Good first issue**

The layout is currently set as a percentage in [here](https://github.com/laurent22/joplin/blob/master/CliClient/app/app-gui.js#L244). The `factor` parameter is a bit like the `flex` parameter: the higher it is, the more space the component takes. In this case, the folder list and note list have a factor of "1" while the note component has a factor of "2". It means the first two components takes each 25% and the note component 50%.

A way to customise this would be to save the layout to setting variables (see models/Setting), for example under "layout.folderList.factor", "layout.noteList.factor" and "layout.note.factor", which would default respectively to "1", "1" and "2".

**Comment by sbussetti** [comment]
> > IWBTAT
> 
> what?
> 
> But otherwise yes that would be a useful feature.

It Would Be Totally Awesome To ...

**Comment by alishamohanty** [comment]
Hey @laurent22 can I work on the issue?

**Comment by coderrsid** [comment]
@laurent22 Yeah sure we can save the different factor settings in model/Setting.js. But how the user would be able to change that easily?

**Comment by pranavmodx** [comment]
@laurent22 I made the changes. But while committing I'm getting lint error 'Mixed spaces and tabs'. I tried all I could but can't get it resolved. Any idea/suggestions?

**Comment by jyuvaraj03** [comment]
I would like to work on this issue. I looked at the closed PR #2613. The way it was implemented in that PR, the changed layout configuration only works when the app is restarted. But, as a user, I would expect the layout to change the moment after I change the layout factor.

If the change has to happen the instant after the layout config is changed, I assume `AppGui.buildUi()` should be called somewhere. Or should we just tell the user to restart the app to see the changes?

Any suggestions?
